[HB-10916] Update roles docs to mention draft permissions (#181)
hashboard-hq/documentation: pages/docs/users-and-permissions/managing-users-and-roles.mdx

  roles, or create a new role that does not have the `READ_RAW_DATA` permission.
</Callout>

<br />

## Configuring roles

> Hashboard uses role based access control to allow users to do specific actions within Hashboard. You can assign system provided roles, configure data access and limit which actions specific users can take inside of Hashboard.

Reviewer: ```suggestion
Hashboard uses role based access control to allow users to do specific actions within Hashboard. You can assign system-provided roles, configure data access, and limit which actions specific users can take inside of Hashboard.
```

<Callout type="warning">
  **Limitations**— Roles are limited today in Hashboard to two main functions:

>   1. Governing which actions can be taken such as creating saved explorations

Reviewer: ```suggestion
  1. Governing which actions can be taken, such as creating saved explorations
```

  and editing dashboards.
  2. Governing access to data models. If you do not have access to read data from

>   a data model, then saved explorations for that data model will not work.

Reviewer: ```suggestion
  a data model, then saved explorations, project metrics, etc., which are derived from that data model will not work.
```

> Today, it is not possible to remove access from a specific saved exploration

Reviewer: ```suggestion
It is not possible to remove access from a specific saved exploration
```

independently from a data model.

</Callout>

### Adding new roles

1. Navigate to the `People` page using the link on the left sidebar.
2. Click `Roles`.
3. Click the `+ New Role` button.
4. Fill out the form and click the submit button: `Create Role`
   - **Name:** The name of the role.
   - **Description:** A description of the role.
5. Your new role should appear in the list of roles. Click the `Edit` button to edit the role.

> 6. Your new role will initially have no permissions rules. Click the `+` button to add a new permission rule. Name your permission rule with a descriptive name.

Reviewer: ```suggestion
6. Your new role will initially have no permission rules. Click the `+` button to add a new permission rule. Name your permission rule with a descriptive name.
```

   <Callout type="info">

>      **What are permission rules?**— Permission rules are a set of permissions

Reviewer: ```suggestion
     **What are permission rules?**— A permission rule is a set of permissions
```

     dashboards in a project. You can also create a separate permission rule
     that allows users with the role to read only a specific data model.
   </Callout>
   <br />{" "}
7. Select resources that the rule will apply to with the `Data Access` section.
   Click the `Add` button to select from a list of all data models as well as a wildcard that targets any data model. Alternatively you can click the `Add all` button to add all models to the rule and the selectively remove models.
8. If you want the role to access data using a specific set of credentials, you can add [role-based database credentials](/docs/database-connections/role-based-credentials) to the rule. These are optional and when omitted, the users of the role will use the default credentials for the data connection.
9. Select permissions that the rule will apply to with the `Permissions` section.
   Click the `Add` button to select from a list of all permissions. Alternatively you can click the `Add all` button to add all permissions to the rule and the selectively remove permissions.
   You can also easily start from the system defined roles by clicking one of additional buttons in the `Permissions` section.

> 10. Select permissions that the will apply when the user is inside a [project draft](/docs/project-management/drafts) with the `Draft Permissions` section.

Reviewer: ```suggestion
10. Select permissions that the rule will apply when the user is inside a [project draft](/docs/project-management/drafts) with the `Draft Permissions` section, if desired.
```
